backend/services/office_hours/oh_event.py

```diff
@@ -0,0 +1,111 @@
+from fastapi import Depends
+from sqlalchemy import select
+from sqlalchemy.orm import Session
+from backend.database import db_session
+from backend.entities.academics.event_entity import EventEntity
+from backend.entities.office_hours.oh_event_entity import OfficeHoursEventEntity
+from backend.models.coworking.time_range import TimeRange
+from backend.models.office_hours.oh_event import OfficeHoursEvent, OfficeHoursEventDraft
+from backend.models.office_hours.oh_event_details import OfficeHoursEventDetails
+from backend.models.user import User
+from backend.services.exceptions import ResourceNotFoundException
+
+from backend.services.permission import PermissionService
+
+
+__authors__ = ["Sadie Amato", "Bailey DeSouza"]
+__copyright__ = "Copyright 2024"
+__license__ = "MIT"
+
+
+class OfficeHoursEventService:
+    """Service that performs all of the actions on the `OfficeHoursEvent` table"""
+
+    def __init__(
+        self,
+        session: Session = Depends(db_session),
+        permission_svc: PermissionService = Depends(),
+    ):
+        """Initializes the database session."""
+        self._session = session
+        self._permission_svc = permission_svc
+
+    
+    def create(self, subject: User, oh_event: OfficeHoursEventDraft) -> OfficeHoursEventDetails:
+        """Creates a new office hours event.
+
+        Args:
+            subject: a valid User model representing the currently logged in User
+            oh_event: OfficeHoursEventDraft to add to table
+
+        Returns:
+            OfficeHoursEventDetails: Object added to table
+        """
+        #TODO
+        return None
+    
+
+    def update(self, subject: User, oh_event: OfficeHoursEvent) -> OfficeHoursEventDetails:
+        """Updates an office hours event.
+
+        Args:
+            subject: a valid User model representing the currently logged in User
+            oh_event: OfficeHoursEvent to update in the table
+
+        Returns:
+            OfficeHoursEventDetails: Updated object in table
+        """ 
+        #TODO
+        return None
+    
+    
+    def delete(self, subject: User, oh_event_id: int) -> None:
+        """Deletes an office hours event.
+
+        Args:
+            subject: a valid User model representing the currently logged in User
+            oh_event_id: ID of office hours event to delete
+        """
+        #TODO
+
+    
+    def get_events_by_section(self, subject: User, oh_section_id: int) -> list[OfficeHoursEventDetails]:
```

Comment:
would this make more sense in the OH section service? it might be more efficient to query office hours sections and return the list of their events (rather than query events and return all that match the section id). i think this is fine too though!!